src/esp/gfx/replay/Recorder.h

```diff
@@ -126,10 +126,23 @@ class Recorder {
                                  bool usePrettyWriter = false);
 
   /**
-   * @brief write saved keyframes to string.
+   * @brief write saved keyframes to string. '{"keyframes": [{...},{...},...]}'
    */
   std::string writeSavedKeyframesToString();
 
+  /**
+   * @brief write saved keyframes as individual strings ['{"keyframe": ...}',
+   * '{"keyframe": ...}', ...]
```

Comment:
Hmm, this just piles on more and more workarounds that were initially caused by the parser expecting a single keyframe to not be a `{...}` but rather `{"keyframe": {...}}`, which then means one then has to process `[{...}, {...}, {...}]` into `[{"keyframe": {...}}, {"keyframe": {...}}, {"keyframe": {...}}]` and it's all a lot more code and work and runtime overhead than if the API would accept just the `{...}` directly.

I remember already partially working around that for the `replayer` utility with adding a [new variant that doesn't require the `"keyframe"` around](https://github.com/facebookresearch/habitat-sim/blob/e6e3cfd1eb7f5822c1b2e95de64059ea32c2902d/src/esp/gfx/replay/Player.cpp#L70-L77) -- could we switch to using that API always everywhere and reduce the two `writeSavedKeyframesToString*()` to be just one, returning either a `[{...}, {...}]` or a list of `{...}`, `{...}`? The former would be a lot less allocations, so preferrable from my side. 

I assume the main reason for this vector-returning API to exist is to avoid a JSON parsing-splitting-serizalization roundtrip on the Python side, right? Separating a `[{...}, {...}]` to individual keyframes as string sub-ranges is doable in an efficient way with no string copies or number parsing for example like in the `replayer` utility below, so the Python bindings could be doing that instead.

https://github.com/facebookresearch/habitat-sim/blob/e6e3cfd1eb7f5822c1b2e95de64059ea32c2902d/src/utils/replayer/replayer.cpp#L112-L126

Reply:
I agree with your suggestions but I'm just being lazy here. I think of all these keyframe APIs as temporary until we do a perf optimization pass and refactor them. We've talked about moving away from JSON altogether.

Reply:
Right, then it doesn't matter that much. I wasn't aware that this was all meant to be temporary.